File: Challenge 16/c16_intermediate.py
'''
    Daily Programmer Challenge 16 : Intermediate
    @author:    Baron Daugherty
    @date:      2015-06-23
    @desc:      Your task is to write a program that simulates a game of craps.

    My crappy craps simulator!
'''

#need random module for die rolls
import random

#main method drives the program
def main():
    #number of rolls, wins/craps for come out
    rolls = 1
    wins = [7, 11]
    crap = [2, 3, 12]

    #roll the come out
    die_roll = roll()

    #check die for win/loss, output if necessary
    if die_roll in wins:
        print("%d wins on the come out!" %die_roll)
    elif die_roll in crap:
        print("%d craps out on the come out!" %die_roll)
    else:
        #set a point and loop to roll for it until win/loss
        print("%d is your point!" %die_roll)
        point = die_roll
        die_roll = 0
        while die_roll != point:
            die_roll = roll()
            rolls += 1
            if die_roll == 7:
                print("%d craps out on the pass!" %die_roll)
                break
            else:
                print("Rolled %d, trying for %d" %(die_roll, point))
        if die_roll == point:
            print("%d wins on the pass!" %die_roll)

        #output total rolls
        print("Total of %d rolls" %rolls)

#roll rolls the bones!
def roll():
    #seed the number generator
    random.seed()

    #deliver us a sum
    return random.randint(1, 6) + random.randint(1, 6)

#start the program
main()

File: Challenge 16/c16_intermediate.pl
#!/usr/bin/perl
#   Daily Programmer Challenge 16 : Easy
#   @author:    Baron Daugherty
#   @date:      2015-07-17
#   @desc:      Your task is to simulate a game of craps.
#
#   A crappy craps simulator!
use warnings;
use strict;

my $num_rolls = 0;
my $point;
my $roll;
my $rolling = 1;

while ($rolling) {
    $roll = int(rand(6)+1) + int(rand(6)+1);
    $num_rolls++;
    
    if ($num_rolls eq 1) {
        $point = $roll;
        if ($point eq 2 or $point eq 3 or $point eq 12) {
           print("Player craps on the come out with $point\n");
           $rolling = 0;
        }
        elsif ($point eq 7 or $point eq 11){
            print("Player wins on the come out with $point\n");
            $rolling = 0;
        }
        else{
            print("Player point set at $point\n");
        }
    }
    else{
        if ($roll eq 7) {
            print("Player craps on the pass with $roll\n");
            $rolling = 0;
        }
        elsif($roll eq $point){
            print("Player wins on the pass! Rolled $point\n");
            $rolling = 0;
        }
        else{
            print("Rolled $roll, trying for $point\n");
        }
    }
}
print("Finished in $num_rolls rolls\n");